src/pocketmine/Player.php

```diff
@@ -2713,8 +2723,27 @@ public function handlePlayerAction(PlayerActionPacket $packet) : bool{
 				}
 				return true;
 			case PlayerActionPacket::ACTION_START_GLIDE:
+				if($this->inventory->getChestplate()->getId() !== Item::ELYTRA){
+					$this->server->getLogger()->warning($this->getName() . " tried to start glide without Elytra");
+					return false;
+				}
+				$ev = new PlayerToggleGlideEvent($this, true);
+				$this->server->getPluginManager()->callEvent($ev);
+				if($ev->isCancelled()){
+					$this->sendData($this);
+				}else{
+					$this->setGliding(true);
+				}
+				return true;
 			case PlayerActionPacket::ACTION_STOP_GLIDE:
-				break; //TODO
+				$ev = new PlayerToggleGlideEvent($this, false);
```

Comment:
So  - no check for elytra here?

Also remember that you can take off / put on the elytra mid flight - which calls the toggle.